src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -2,75 +2,68 @@
-            null,   // Empty 
-            null,   // Object
-            null,   // DBNull
-            new JsonValueConverterBoolean(),
-            new JsonValueConverterChar(),
-            new JsonValueConverterSByte(),
-            new JsonValueConverterByte(),
-            new JsonValueConverterInt16(),
-            new JsonValueConverterUInt16(),
-            new JsonValueConverterInt32(),
-            new JsonValueConverterUInt32(),
-            new JsonValueConverterInt64(),
-            new JsonValueConverterUInt64(),
-            new JsonValueConverterSingle(),
-            new JsonValueConverterDouble(),
-            new JsonValueConverterDecimal(),
-            new JsonValueConverterDateTime(),
-            null,   // (not a value)
-            new JsonValueConverterString()
-        };
-
-        private static readonly object[] s_NullableConverters = new object[MaxTypeCode + 1]
+        internal static object Create(Type type)
         {
-            null,
-            null,
-            null,
-            new JsonValueConverterBooleanNullable(),
-            new JsonValueConverterCharNullable(),
-            new JsonValueConverterSByteNullable(),
-            new JsonValueConverterByteNullable(),
-            new JsonValueConverterInt16Nullable(),
-            new JsonValueConverterUInt16Nullable(),
-            new JsonValueConverterInt32Nullable(),
-            new JsonValueConverterUInt32Nullable(),
-            new JsonValueConverterInt64Nullable(),
-            new JsonValueConverterUInt64Nullable(),
-            new JsonValueConverterSingleNullable(),
-            new JsonValueConverterDoubleNullable(),
-            new JsonValueConverterDecimalNullable(),
-            new JsonValueConverterDateTimeNullable(),
-            null,
-            new JsonValueConverterString()
-        };
-
-        internal static object GetDefaultPropertyValueConverter(Type propertyType, bool isNullable)
-        {
-            object converter = null;
-
-            int typeCode = (int)Type.GetTypeCode(propertyType);
-            if (typeCode <= MaxTypeCode)
+            if (type.IsEnum)
             {
-                if (isNullable)
-                {
-                    converter = s_NullableConverters[typeCode];
-                }
-                else
-                {
-                    converter = s_Converters[typeCode];
-                }
+                return Activator.CreateInstance(
+                    typeof(JsonValueConverterEnum<>).MakeGenericType(type),
+                    BindingFlags.Instance | BindingFlags.NonPublic,
+                    binder: null,
+                    new object[] { false },
+                    culture: null);
             }
-
-            return converter;
+            else if (type == typeof(Enum))
+            {
+                return new JsonValueConverterEnum(false);
+            }
+            TypeCode typeCode = Type.GetTypeCode(type);
+            switch (typeCode)
```

Comment:
Why change from array to `switch`?

Reply:
This `Create` method is only ran once per `Type` and is called only from the generic `DefaultConverters<TValue>.s_converter` initializer. There's no need to store the converter type in an array when it's being stored in this generic static field.